DSpace/dspace-angular
# Browse links and "by regex" links in item display (Angular)

## References
Related to [DSpace#8589](https://github.com/DSpace/DSpace/pull/8589)
Related to [REST Contract#205](https://github.com/DSpace/RestContract/pull/205)

## Description
This change allows the following configurable display of links in item page metdata values:
1. The configuration in `webui.browse.links.<n>` (dspace.cfg) now works as it did in previous versions, where a browse link is rendered for the metadata value in question if it is configured for a valid browse index. This works for both plain item page field components, and for metadata representations where there is virtual metadata instead of a link to a related entity. (related entity link will always take precedence)
3. The item page field components (and children) now allow a `[urlRegex]` declarative in the templates where a link will be rendered if and only if the regex matches the metadata text value, eg. `^https?://`, or something more restrictive

## To test conditional regex-based linking:

Try adding a section to a typed or untyped item template with a field like:
```
<ds-generic-item-page-field [item]="object"
      [fields]="['dc.subject']"
      [separator]="', '"
      [label]="'item.page.subject'">
      [urlRegex]="'^https?://www.something.com'"
    </ds-generic-item-page-field>
```

And try it out with different subject values, some of which match the regular expression.

## To test browse links:

Set up browse link config for fields, such as:
```
webui.browse.link.1 = author:dc.contributor.*
webui.browse.link.2 = subject:dc.subject.*
```

And try out items with values in these fields. With the author field, note that if it is not a "authorOf..." relation, the browse author link should be displayed. If it *is* a relation to a person entity, it should instead be linked to that entity.

## Checklist
_This checklist provides a reminder of what we are going to look for when reviewing your PR. You need not complete this checklist prior to creating your PR (draft PRs are always welcome). If you are unsure about an item in the checklist, don't hesitate to ask. We're here to help!_
